server/src/main/java/org/elasticsearch/cluster/metadata/ComposableIndexTemplate.java

```diff
@@ -256,35 +259,63 @@ public String toString() {
 
     public static class DataStreamTemplate implements Writeable, ToXContentObject {
 
+        public static final Version DATA_STREAM_ALIAS_VERSION = Version.V_8_0_0;
+
         private static final ParseField HIDDEN = new ParseField("hidden");
+        private static final ParseField ALIASES = new ParseField("aliases");
 
         public static final ConstructingObjectParser<DataStreamTemplate, Void> PARSER = new ConstructingObjectParser<>(
             "data_stream_template",
             false,
-            a -> new DataStreamTemplate(a[0] != null && (boolean) a[0]));
+            args -> {
+                boolean hidden = args[0] != null && (boolean) args[0];
+                @SuppressWarnings("unchecked")
+                Map<String, AliasMetadata> aliases = (Map<String, AliasMetadata>) args[1];
+                return new DataStreamTemplate(hidden, aliases);
+            }
+        );
 
         static {
             PARSER.declareBoolean(ConstructingObjectParser.optionalConstructorArg(), HIDDEN);
+            PARSER.declareObject(ConstructingObjectParser.optionalConstructorArg(), (p, c) -> {
+                Map<String, AliasMetadata> aliasMap = new HashMap<>();
+                while ((p.nextToken()) != XContentParser.Token.END_OBJECT) {
+                    AliasMetadata alias = AliasMetadata.Builder.fromXContent(p);
+                    aliasMap.put(alias.alias(), alias);
+                }
+                return aliasMap;
+            }, ALIASES);
         }
 
         private final boolean hidden;
 
+        @Nullable
+        private final Map<String, AliasMetadata> aliases;
```

Comment:
I think we can reuse the alias definitions in `ComposableIndexTemplate` (defined inside the `Template` class)?
If a composable index template has a `DataStreamTemplate` and aliases definitions are present then data stream aliases are created. If just alias definitions are present in composable index template then indices aliases are created.